src/sid/update_states.py

```diff
@@ -156,7 +161,7 @@ def _update_info_on_new_tests(
     # For everyone who received a test result, the countdown for the test processing
     # has expired. If you have a positive test result (received_test_result &
     # immune) you will leave the state of knowing until your immunity expires.
-    states["new_known_case"] = states["received_test_result"] & states["immune"]
+    states["new_known_case"] = states["received_test_result"] & states["immunity"] > 0
```

Comment:
Unfortunately I think this (``& states["immunity"] > 0 ``) is wrong, since the immunity level may be non-zero for various reasons. I do not know what the correct expression is with the new non-binary setting.

Reply:
Even better:

```suggestion
    states["new_known_case"] = states["received_test_result"] & (states["infectious"]  | states["symptomatic"])
```